### PR TITLE
feat(registry): add SN31 Recall TaoMarketCap subnet-api surface

### LINKED ISSUE #4032: Enrich SN31 Recall — add public API endpoint
Part of #427.

**Subnet 31 — Recall** is missing from the registry: **public API endpoint**. Research whether the subnet exposes this, find the real public link, and submit it.

### Find it
Search for the subnet's official public API endpoint (project site / GitHub / docs / Bittensor). Confirm the link is real, public, no-auth, and genuinely SN31's — cross-reference the subnet name + Bittensor. ⚠️ Many subnets simply don't expose a public API yet, and on-chain identity links are often **stale/dead** — verify the link actually resolves before submitting. If the subnet genuinely exposes no such public surface, comment here so a maintainer can close it.

### Submit — one subnet, one file
Surfaces live in **one file per subnet**: `registry/subnets/<slug>.json` → its `surfaces[]` array. First find the provider slug for the team/operator behind the surface (a wrong slug is the #1 validation failure): `npm run providers:list`. If the subnet has no manifest yet, scaffold it first with `npm run subnet:new -- --netuid 31 --write`. Then append the surface to the subnet's file:
```bash
npm run surface:add -- --netuid 31 --kind subnet-api \
  --url <real-public-url> --source-url <link-that-proves-it> \
  --provider <slug> --submitted-by <your-login> --write
```
`surface:add` writes it with `authority: "community"` and `review.state: "community-submitted"`.

Open a PR touching **exactly one** `registry/subnets/<slug>.json` file — append-only for an existing manifest, or a `subnet:new` scaffold plus surface for a missing manifest. The review gate validates + reviews it. (The per-candidate-file lane is retired: recreating `registry/candidates/community/*.json` is rejected by CI.)

**Rules:** a real `url` that resolves · a `source_url` that proves it's official · one file · no generated artifacts · `public_safe: true` · `auth_required: false`. Full guide: [CONTRIBUTING → Community submissions](https://github.com/JSONbored/metagraphed/blob/main/CONTRIBUTING.md#community-submissions).

### PR DESCRIPTION
**Pull request by philluiz2323**
Closes #4032

## What
SN31 Recall (`registry/subnets/recall.json`) currently has zero registered surfaces — the maintainer's reviewed overlay confirms no official website, source repository, API, OpenAPI schema, or docs site is verified for this subnet. TaoMarketCap's public developer API independently indexes on-chain subnet state for every netuid regardless of whether the subnet team publishes anything themselves, and this is already the established pattern used for other zero-first-party-presence subnets in this registry (e.g. Academia SN109, Endure Network SN30).

## Proof it's real
- `GET https://api.taomarketcap.com/public/v1/subnets/31` resolves (301 → `.../31/`) and returns live JSON: `subnetName: "rec4ll"`, a description matching Recall's known RAG/retrieval-augmented-generation positioning ("Miners serve embedding models, vector search, and LLM inference. Validators independently evaluate retrieval accuracy..."), plus on-chain economics/hyperparameters and dTAO market fields — cross-referencing correctly to netuid 31.
- `source_urls` cite TaoMarketCap's own public developer documentation plus `https://backprop.finance/dtao/subnets/31-recall`, an independent public page that also identifies this netuid as Recall/RAG, corroborating it's genuinely SN31's data, not a misattributed netuid.

## Classification note
An earlier, unrelated PR for a different subnet was closed because this same TaoMarketCap `/public/v1/subnets/{netuid}` endpoint was misclassified as `kind: "data-artifact"` when the reviewer flagged it should be `kind: "subnet-api"` (a live, parameterized, callable REST endpoint — not a static dataset). This PR uses `kind: "subnet-api"` from the start, matching that correction and the convention already used for the Academia/Endure Network entries.

## Changes
- **`registry/subnets/recall.json`** — appends exactly one surface to the (previously empty) `surfaces[]` array via `npm run surface:add`: `id: "sn-31-taomarketcap-subnet-api"`, `kind: "subnet-api"`, `authority: "community"`, `review.state: "community-submitted"`, `auth_required: false`, `public_safe: true`. No other field in the file is touched.

## Acceptance criteria
- [x] Exactly one file changed: `registry/subnets/recall.json`
- [x] `url` resolves and returns real, live SN31 data
- [x] `source_urls` independently corroborate it's genuinely SN31's data
- [x] `authority: "community"`, `review.state: "community-submitted"`
- [x] `auth_required: false`, `public_safe: true`
- [x] No secrets, no private/localhost URLs

## Testing
- `npm run validate:surface -- registry/subnets/recall.json`: "Surface validation passed: 1 surface(s) across 1 subnet file(s)."
- `npm run scan:public-safety`: "Public-safety scan passed."

### CODE REVIEW

**Comment by philluiz2323** [comment]
For anyone reviewing: this content was genuine and verified (real, live TaoMarketCap subnet-api data for SN31, cross-referenced against an independent source, classified as `subnet-api` per the existing repo convention). CI failed because `tests/artifacts.test.mjs` hardcodes netuid 31 as its canonical 'zero-surface / blocked agent-readiness' fixture across several assertions (lines ~983-992, ~1070-1077, ~1087-1092) — adding SN31's first surface correctly flips its computed agent-readiness from blocked to callable, which the test doesn't expect. Fixing that fixture would require touching the test file, out of scope for a single-file registry PR. Withdrawing rather than fighting the test.
